examples/specs_integration/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+#[macro_use]
+extern crate gdnative as godot;
+#[macro_use]
+extern crate lazy_static;
+
+extern crate euclid;
+extern crate nalgebra;
+extern crate specs;
+
+pub mod helper;
+pub mod singleton;
+pub mod sync;
+pub mod your_state;
+
+use specs::prelude::*;
+
+use godot::{GodotString, Variant};
+use nalgebra::geometry::{Translation3, UnitQuaternion};
+use nalgebra::Isometry3;
+use singleton::try_get_singleton_state;
+use sync::ExampleScene::ExampleCube;
+use sync::{GdSceneLoc, GodotStateSync, Pos, Template3dName};
+use your_state::ExampleComponent;
+
+#[derive(gdnative::NativeClass)]
+#[inherit(gdnative::Spatial)]
+struct SpecsIntegration {
+    // Store the loaded scene for a very slight performance boost but mostly to show you how.
+    children_spawned: u32,
+    state_sync: GodotStateSync, // Tracks stuff created in Godot in the past
+}
+
+// Assume godot objects are safe to Send
+unsafe impl Send for SpecsIntegration {}
+
+// Demonstrates the usage of Specs to manage a scene. UI / interactions based on scene_create
+//
+//   0. Every frame, the contents of the Specs World (Components & Entities) are synced to Godot
+//
+//  -- This stuff is similar to scene_create
+//   1. Component in Specs is created when spawn_one is called
+//   2. Specs Components are deleted when remove_one is called
+//   3. Find and call functions in a node (Panel)
+//   4. Call functions in GDNative (from panel into spawn/remove)
+//
+//  Note, the same mechanism which is used to call from panel into spawn_one and remove_one can be
+//   used to call other GDNative classes here in rust.
+
+#[gdnative::methods]
+impl SpecsIntegration {
+    fn _init(_owner: gdnative::Spatial) -> Self {
+        SpecsIntegration {
+            children_spawned: 0,
+            state_sync: GodotStateSync::new(),
+        }
+    }
+
+    #[export]
+    fn _ready(&mut self, _owner: gdnative::Spatial) {
+        self.state_sync.register_scene(
+            Template3dName::ExampleScenes(ExampleCube),
+            "res://Child_scene.tscn",
+        );
+    }
+
+    #[export]
+    fn _process(&mut self, owner: gdnative::Spatial, _delta: f64) {
+        self.state_sync.do_sync(owner, |_world| {
+            // This is a handy place where you have access to the world right before we
+            //  synchronise it. Perhaps run your own systems here?
+        });
+    }
+
+    #[export]
+    unsafe fn spawn_one(&mut self, mut owner: gdnative::Spatial, message: GodotString) {
+        godot_print!("Called spawn_one({})", message.to_string());
+
+        // Note - This will fail if you call try_get_singleton_state from inside other code that already
+        //   holds this mutex...
+        let mut your_state = if let Some(state) = try_get_singleton_state() {
+            state
+        } else {
+            return;
+        };
```

Comment:
Perhaps use `expect` instead of failing silently? This is an example and methods are wrapped in `catch_unwind` so it should be fine here.